From 901ae1f0db14fd339f6064f8d4c5303e3dade8ad Mon Sep 17 00:00:00 2001
From: Richard Woodruff <r-woodruff2@ti.com>
Date: Wed, 16 Mar 2011 16:41:32 -0500
Subject: [PATCH] OMAP4: Extending SO memory attributes

Memory attributes for areas used by both processors should be marked
as shared

Change-Id: I3e038bd38c59f83679061d6352c8fa4394eaee34
Signed-off-by: Carlos Leija <cileija@ti.com>
---
 arch/arm/mm/mmu.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: kernel/arch/arm/mm/mmu.c
===================================================================
--- kernel.orig/arch/arm/mm/mmu.c	2011-05-13 12:17:23.000000000 +0900
+++ kernel/arch/arm/mm/mmu.c	2011-05-13 12:17:27.000000000 +0900
@@ -263,13 +263,13 @@
 
 	[MT_MEMORY_SO] = {
 		.prot_sect = PMD_TYPE_SECT | PMD_SECT_AP_WRITE |
-				PMD_SECT_UNCACHED | PMD_SECT_XN,
+				PMD_SECT_UNCACHED | PMD_SECT_XN | PMD_SECT_S,
 		.domain    = DOMAIN_KERNEL,
 	},
 
 	[MT_MEMORY_SO_EXE] = {
 		.prot_sect = PMD_TYPE_SECT | PMD_SECT_AP_WRITE |
-				PMD_SECT_UNCACHED,
+				PMD_SECT_UNCACHED | PMD_SECT_S,
 		.domain    = DOMAIN_KERNEL,
 	},
 };
